addons/survey/models/survey_question.py

```diff
@@ -265,6 +265,19 @@ def _compute_is_scored_question(self):
             else:
                 question.is_scored_question = False
 
+    # ------------------------------------------------------------
+    # CRUD
+    # ------------------------------------------------------------
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_live_sessions_in_progress(self):
+        for survey_id in self.survey_id:
```

Comment:
Why not doing a batch version, like
```
running_surveys = self.survey_id.filtered(lambda survey: survey.session_state == 'in_progress')
if running_surveys:
    raise UserError(_(
        'You cannot delete questions from surveys "%(survey°names)s" while live sessions are in progress.',
        survey_names=', '.join(running_surveys.mapped('title')),
    ))
```

Reply:
Only to spare development resources as this is probably an even rarer case, but indeed it's even more convenient for the user, if you can squeeze in something like this @sami-odoo .

Reply:
Hello @flch-odoo , @tde-banana-odoo 
Requested chnges are done . Please review it !